aws/data_source_aws_db_subnet_group_test.go

```diff
@@ -0,0 +1,59 @@
+package aws
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform/helper/acctest"
+	"github.com/hashicorp/terraform/helper/resource"
+)
+
+func TestAccAWSDbSubnetGroupDataSource_basic(t *testing.T) {
+	rInt := acctest.RandInt()
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:  func() { testAccPreCheck(t) },
+		Providers: testAccProviders,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAWSDBSubnetGroupDataSourceConfig(rInt),
+				Check: resource.ComposeAggregateTestCheckFunc(
```

Comment:
here we should follow the pattern of using "TestCheckResoureAttrPair() to verify values in the Terraform state for data sources attributes to compare them with their expected resource attributes"..in the cases where the data_source has an attribute different from the resource (e.g. vpc_id and status), using `resource.TestCheckResourceAttrSet` works as is.